src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -1446,6 +1446,36 @@ protected theorem lt_of_le_ne (x y : BitVec n) (h1 : x <= y) (h2 : ¬ x = y) : x
   simp
   exact Nat.lt_of_le_of_ne
 
+/-! ### intMin -/
+
+/-- The bitvector of width `w` that has the smallest value when interpreted as an integer. -/
+def intMin (w : Nat) : BitVec w := BitVec.ofNat w (2^(w - 1))
+
+theorem getLsb_intMin (w : Nat) : (intMin w).getLsb i = decide (i + 1 = w) := by
+  simp only [getLsb, intMin, toNat_ofNat, Nat.testBit_mod_two_pow]
+  rcases Nat.lt_trichotomy (i + 1) w with lt | rfl | gt
+  case inl =>
+    rw [Nat.testBit_two_pow_of_ne (by omega)]
+    simp
+    omega
+  case inr.inl =>
+    simp
+  case inr.inr =>
+    simp [show ¬ (i + 1 = w) by omega]
+    omega
```

Comment:
I'm not super happy with this. I think `intMin` and `intMax` should be moved below `twoPow`.

Then use `def intMin (w : Nat) : BitVec w := twoPow w (w - 1)`, and `def intMax (w : Nat) : BitVec w := allOnes w`.

In fact both could be abbrevs, or just not get defined?

Every new def adds lots of missing API which is not keeping up.

Reply:
Moreover, the proof of `getLsb_intMin` should be `by simp [intMin, boolToPropSimps]; omega`, for a simp set that we don't have yet that converts statements about `decide` into the corresponding propositional formulas.

Reply:
Done.